narwhals/_pandas_like/dataframe.py

```diff
@@ -335,6 +318,23 @@ def generate_unique_token(
                     ),
                 )
 
+        if how == "anti":
+            indicator_token = generate_unique_token(
+                n_bytes=8, columns=[*self.columns, *other.columns]
+            )  # pragma: no cover
+
+            return self._from_dataframe(
+                self._dataframe.merge(
+                    other._dataframe.loc[:, left_on],
```

Comment:
selecting only the join columns in here so that no additional columns get created